sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java

```diff
@@ -395,11 +395,7 @@ public static <T> Flux<? extends IDocumentQueryExecutionContext<T>> createSpecia
                 int pageSizeWithTopOrLimit = Math.min(queryInfo.hasTop() ? queryInfo.getTop() : Integer.MAX_VALUE,
                                            queryInfo.hasLimit() && queryInfo.hasOffset() ?
                                                queryInfo.getLimit() + queryInfo.getOffset() : Integer.MAX_VALUE);
-                if (initialPageSize > 0) {
-                    initialPageSize = Math.min(pageSizeWithTopOrLimit, initialPageSize);
-                } else {
-                    initialPageSize = pageSizeWithTopOrLimit;
-                }
+                initialPageSize = pageSizeWithTopOrLimit;
```

Comment:
why the change here?

Reply:
Because for Top greater than 100, and maxItemCount not set, initialPageSize by default is 100. So the top value was always being 100. 

And if Non-streaming order by, top or limit will always be there, and their value should be picked up.

Reply:
limit is dangerous - because for several queries offset/limit are actually client-side processing

Reply:
But `initialPageSize` is used as the size of our PQ. For limit if we use the default `initialPageSize`, the we will not get the correct amount of items back.

Reply:
From some of the discussions on the JS SDK, I think we need to have `OFFSET+LIMIT` tests for at least the single partition and multiple partition cases, since they are processed differently by the service.